Sources/Action.swift

```diff
@@ -148,6 +147,16 @@ public protocol ActionProtocol {
 	/// `NoError` can be used.
 	associatedtype Error: Swift.Error
 
+	/// Initializes an action that will be conditionally enabled, and creates a
+	/// SignalProducer for each input.
+	///
+	/// - parameters:
+	///   - enabledIf: Boolean property that shows whether the action is
+	///                enabled.
+	///   - execute: A closure that returns the signal producer returned by
+	///              calling `apply(Input)` on the action.
+	init<P: PropertyProtocol>(enabledIf property: P, _ execute: @escaping (Input) -> SignalProducer<Output, Error>) where P.Value == Bool
```

Comment:
I tried making this change (having the execute closure return `SignalProducer<Output, ActionError<Error>>`), but I couldn't figure out a nice way to make the API work without breaking the existing use case.

However, I tried making the return value an optional signal producer, which seems to work nicely: https://github.com/ReactiveCocoa/ReactiveSwift/commit/c8791b7a75123e4885262b7da5cdbc5f7bb8668f. 

There's something still wrong to me about this though. I think if we want to truly prove this is safe, we need to make checking the enabled state and grabbing the input value an atomic operation, which means moving the input property into `ActionState` somehow...